File: jchirp/src/main/java/com/moshy/jchirp/responses/ChirpResponse.java
package com.moshy.jchirp.responses;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.time.OffsetDateTime;
import java.util.UUID;

@Data
public class ChirpResponse {

    private UUID id;
    @JsonProperty("created_at")
    private OffsetDateTime createdAt;
    @JsonProperty("updated_at")
    private OffsetDateTime updatedAt;

    private String body;

    @JsonProperty("user_id")
    private UUID userId;
}

File: chirpkt/src/main/kotlin/chirp/ChirpResponse.kt
package chirp

import kotlinx.datetime.Instant
import kotlinx.serialization.*
import java.util.UUID

@Serializable
internal data class ChirpResponse(
    @Serializable(with = UUIDSerializer::class)
    val id: UUID,
    @SerialName("created_at")
    val created: Instant,
    @SerialName("updated_at")
    val updated: Instant,
    val body: String,
    @Serializable(with = UUIDSerializer::class)
    val userId: UUID,
)